src/main/java/com/zachary_moore/runner/LintRunner.java

```diff
@@ -5,78 +5,74 @@
 import com.zachary_moore.lint.LintRegister;
 import com.zachary_moore.lint.LintRule;
 import com.zachary_moore.objects.JSONFile;
-
 import java.io.File;
 import java.io.IOException;
-import java.util.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Logger;
 import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+
 
 public class LintRunner {
 
-    private List<JSONFile> filesToLint;
+    private String[] basePaths;
     private Map<LintRule, Map<JSONFile, List<String>>> lintOutput;
     private final LintRegister lintRegister;
 
     /**
      * Create a LintRunner
      * @param lintRegister representation of {@link LintRule} to run against
-     * @param basePath Base Path to setup {@link JSONFile} from
+     * @param basePaths Paths to folders or files to fetch {@link JSONFile}(s) from
      */
     public LintRunner(LintRegister lintRegister,
-                      String basePath) {
+                      String... basePaths) {
         this.lintRegister = lintRegister;
-
```

Comment:
Moved most of this logic to `getAllFiles`. Failing to create an instance because one of the paths is wrong is not very good. Also constructors should not have complicated logic.